### PR TITLE
refactor(availability): merging cache.ShareAvailability and light.ShareAvailability

### LINKED ISSUE #1119: node: Remove `share.CacheAvailability` wrapping over `share.FullAvailability`

### PR DESCRIPTION
**Pull request by distractedm1nd**
Closes #1119

### CODE REVIEW

**Comment by distractedm1nd** [comment]
Crossposting from slack: I’m not even sure why cacheavailability is necessary? The only duplicate hash is the empty hash which short circuits, and the rest of the calls go through the blockservice without touching availability, right?

File: go.mod [hunk 1]
```diff
@@ -55,7 +55,7 @@ require (
 	github.com/spf13/cobra v1.7.0
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.8.4
-	github.com/tendermint/tendermint v0.34.28
+	github.com/tendermint/tendermint v0.35.9
```
Comment (renaynay) [inline go.mod:58]: ???

**Review by renaynay — COMMENTED** [review]
borked units

File: go.mod (hunk 1 above)
Comment (distractedm1nd) [reply go.mod:58]: ??? Also dont know how this happened, thx

**Review by renaynay — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Pull request merged**